workflow-templates/job.build.yml

```diff
@@ -0,0 +1,63 @@
+name: Build Docker Image with Kaniko
+
+on:
+  workflow_call:
+    inputs:
+      aws_region:
+        required: true
+        type: string
+      service:
+        required: true
+        type: string
+      image_tag:
+        required: true
+        type: string
+      dockerfile:
+        required: false
+        type: string
+        default: Dockerfile
+      build_args:
+        required: false
+        type: string
+      context:
+        required: false
+        type: string
+        default: "."
+      cache:
+        required: false
+        type: boolean
+        default: false
+        
```

Comment:
Let's also add an optional param for target `platform` for the container packaging instead of relying on the Runner architecture.
it should default to our current nodes